src/pybind/mgr/dashboard/tests/test_versioning.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+
+from . import ControllerTestCase
+from ..controllers import ApiController, RESTController
+
+
+@ApiController("/vtest", secure=False)
+class VTest(RESTController):
+    RESOURCE_ID = "vid"
+
+    def list(self):
+        return {'version': ""}
+
+    def get(self):
+        return {'version': ""}
+
+    @RESTController.Collection('GET', version="1.0")
+    def vmethod(self):
+        return {'version': '1.0'}
+
+    @RESTController.Collection('GET', version="2.0")
+    def vmethodv2(self):
+        return {'version': '2.0'}
+
+
+class RESTVersioningTest(ControllerTestCase):
+    @classmethod
+    def setup_server(cls):
+        cls.setup_controllers([VTest], "/test")
+
+    def test_v1(self):
+        self._get('/test/api/vtest/vmethod', version="1.0")
+        self.assertStatus(200)
+        self.assertJsonBody({'version': '1.0'})
+
+    def test_v2(self):
+        self._get('/test/api/vtest/vmethodv2', version="2.0")
+        self.assertStatus(200)
+        self.assertJsonBody({'version': '2.0'})
+
+    def test_v1_v2(self):
+        self._get('/test/api/vtest/vmethod', version="2.0")
+        self.assertStatus(415)
+
+    def test_v2_v1(self):
+        self._get('/test/api/vtest/vmethodv2', version="1.0")
+        self.assertStatus(415)
```

Comment:
Same as with the QA tests, you can use subTest here to get parameterized tests and reduced code.